lib/OpenQA/Worker/Jobs.pm

```diff
@@ -302,7 +308,7 @@ sub read_json_file {
     my $fn = "$pooldir/testresults/$name";
     my $ret;
     local $/;
-    open(my $fh, '<', $fn) or return undef;
+    open(my $fh, '<', $fn) or die "can't open $fn: $!";
```

Comment:
Won't this potentially trigger dead workers when i.e. canceling job before os-autoinst fully initialize? Are all json files guaranteed to be present to warrant this death order?